src/ostree/ot-builtin-commit.c

```diff
@@ -600,7 +602,32 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
   if (opt_link_checkout_speedup && !ostree_repo_scan_hardlinks (repo, cancellable, error))
     goto out;
 
-  mtree = ostree_mutable_tree_new ();
+  if (opt_base)
+    {
+      g_autofree char *base_commit = NULL;
+      g_autoptr(GFile) root = NULL;
+      if (!ostree_repo_read_commit (repo, opt_base, &root, &base_commit, cancellable, error))
+        goto out;
+      OstreeRepoFile *rootf = (OstreeRepoFile*) root;
+
+      mtree = ostree_mutable_tree_new_from_checksum (repo,
+                                                     ostree_repo_file_tree_get_contents_checksum (rootf),
+                                                     ostree_repo_file_tree_get_metadata_checksum (rootf));
+
+      if (opt_selinux_policy_from_base)
```

Comment:
Let's also reflect that in the help string for `selinux-policy-from-base`. E.g. something like:

> Set SELinux labels based on --base or otherwise the first --tree argument

?